File: core/src/main/java/com/egoal/darkestpixeldungeon/actors/buffs/ViewMark.java
package com.egoal.darkestpixeldungeon.actors.buffs;

import com.egoal.darkestpixeldungeon.messages.Messages;
import com.egoal.darkestpixeldungeon.ui.BuffIndicator;
import com.watabou.utils.Bundle;

/**
 * Created by 93942 on 5/12/2018.
 */

public class ViewMark extends FlavourBuff{
	
	public int observer	=	0;

	private static final String OBSERVER    = "observer";

	@Override
	public void storeInBundle( Bundle bundle ) {
		super.storeInBundle( bundle );
		bundle.put( OBSERVER, observer );

	}

	@Override
	public void restoreFromBundle( Bundle bundle ) {
		super.restoreFromBundle( bundle );
		observer	=	bundle.getInt( OBSERVER );
	}
	
	@Override
	public int icon(){ return BuffIndicator.STARE; }

	@Override
	public String toString() {
		return Messages.get(this, "name");
	}

	@Override
	public String desc() {
		return Messages.get(this, "desc");
	}
	
}

File: core/src/main/java/com/egoal/darkestpixeldungeon/actors/buffs/ViewMark.kt
package com.egoal.darkestpixeldungeon.actors.buffs

import com.egoal.darkestpixeldungeon.messages.Messages
import com.egoal.darkestpixeldungeon.ui.BuffIndicator
import com.watabou.utils.Bundle

/**
 * Created by 93942 on 5/12/2018.
 */

class ViewMark : FlavourBuff() {

    var observer = 0

    init {
        type = Buff.buffType.NEGATIVE
    }

    override fun storeInBundle(bundle: Bundle) {
        super.storeInBundle(bundle)
        bundle.put(OBSERVER, observer)

    }

    override fun restoreFromBundle(bundle: Bundle) {
        super.restoreFromBundle(bundle)
        observer = bundle.getInt(OBSERVER)
    }

    override fun icon(): Int {
        return BuffIndicator.STARE
    }

    override fun toString(): String {
        return Messages.get(this, "name")
    }

    override fun desc(): String {
        return Messages.get(this, "desc")
    }

    companion object {

        private val OBSERVER = "observer"
    }

}
